pipeline_dp/aggregate_params.py

```diff
@@ -113,45 +113,83 @@ def metrics_str(self) -> str:
+        return self.min_sum_per_partition is not None and self.max_sum_per_partition is not None
+
     def __post_init__(self):
         if self.low is not None:
             raise ValueError(
                 "AggregateParams: please use min_value instead of low")
         if self.high is not None:
             raise ValueError(
                 "AggregateParams: please use max_value instead of high")
-        if self.metrics:
-            needs_min_max_value = Metrics.SUM in self.metrics \
-                                  or Metrics.MEAN in self.metrics \
-                                  or Metrics.VARIANCE in self.metrics
-            if needs_min_max_value and (self.min_value is None or
-                                        self.max_value is None):
-                raise ValueError(
-                    "AggregateParams: min_value and max_value must be set")
-            if needs_min_max_value and (_not_a_proper_number(self.min_value) or
-                                        _not_a_proper_number(self.max_value)):
-                raise ValueError(
-                    "AggregateParams: min_value and max_value must be both "
-                    "finite numbers")
-            if needs_min_max_value and self.max_value < self.min_value:
-                raise ValueError(
-                    "AggregateParams: max_value must be equal to or greater than"
-                    " min_value")
-            if Metrics.VECTOR_SUM in self.metrics and \
-            (Metrics.SUM in self.metrics or \
-            Metrics.MEAN in self.metrics or \
-            Metrics.VARIANCE in self.metrics):
+        if (self.min_value is None) != (self.max_value is None):
+            raise ValueError(
+                "AggregateParams: min_value and max_value should be or both set or both None."
+            )
+        if (self.min_sum_per_partition is None) != (self.max_sum_per_partition
+                                                    is None):
+            raise ValueError(
+                "AggregateParams: min_sum_per_partition and max_sum_per_partition should be or both set or both None."
+            )
+
+        value_bound = self.min_value is not None
+        if value_bound:
+            self._check_range_correctness("min_value", "max_value")
+            if self.max_contributions_per_partition is None:
                 raise ValueError(
-                    "AggregateParams: vector sum can not be computed together "
-                    "with scalar metrics, like sum, mean etc")
+                    "AggregateParams: max_contributions_per_partition must be "
+                    "set when min_value/max_value are set.")
+
+        partition_bound = self.min_sum_per_partition is not None
+        if partition_bound:
+            self._check_range_correctness("min_sum_per_partition",
+                                          "max_sum_per_partition")
+
+        if value_bound and partition_bound:
```

Comment:
I would move this before `if value_bound: check_range_correctness` since if you supply all 4 parameters (but incorrectly), you'll first get an error for that the range is not correct; then after you fix it you'll get an error that you can't supply value and partition bound at the same time.